### ISSUE TITLE
[LLM Runner] Implement GrammarLogitProcessor using llguidance

### ISSUE THREAD

**Issue by kirklandsign** [pytorch/executorch#19475]
Wrap LlgMatcher as LogitProcessor. compute_mask -> apply bitmask to logits -> sample -> consume_token. Support json_schema/regex/lark/gbnf. Depends on: LogitProcessor interface, tokenizer bridge, llguidance dep.

cc @larryliu0820 @mergennachin @cccclai @helunwencser @jackzhxng
